platform/linux-generic/odp_crypto_ipsecmb.c

```diff
@@ -0,0 +1,1063 @@
+/* Copyright (c) 2014-2018, Linaro Limited
+ * Copyright (c) 2021, ARM Limited
+ * Copyright (c) 2022, Nokia
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#include <odp_posix_extensions.h>
+#include <odp/autoheader_internal.h>
+
+#include <odp/api/crypto.h>
+#include <odp/api/spinlock.h>
+#include <odp/api/sync.h>
+#include <odp/api/debug.h>
+#include <odp/api/align.h>
+#include <odp/api/shared_memory.h>
+#include <odp/api/hints.h>
+#include <odp/api/random.h>
+
+#include <odp/api/plat/event_inlines.h>
+#include <odp/api/plat/packet_inlines.h>
+#include <odp/api/plat/queue_inlines.h>
+#include <odp/api/plat/thread_inlines.h>
+
+#include <odp_debug_internal.h>
+#include <odp_global_data.h>
+#include <odp_init_internal.h>
+#include <odp_packet_internal.h>
+
+#include <ipsec-mb.h>
+
+#define MAX_SESSIONS 4000
+/* Length in bytes */
+#define ARM_CRYPTO_MAX_CIPHER_KEY_LENGTH      32
+#define ARM_CRYPTO_MAX_AUTH_KEY_LENGTH        32
+#define ARM_CRYPTO_MAX_IV_LENGTH              16
+#define ARM_CRYPTO_MAX_DATA_LENGTH	      65536
+#define ARM_CRYPTO_MAX_DIGEST_LENGTH          16
```

Comment:
I think the ARM_CRYPTO prefix of these macros could be dropped since the code is not ARM specific.

Reply:
Rename the prefix from ARM_CRYPTO to IPSEC_MB_CRYPTO.